proposals/0281-visible-forall.rst

```diff
@@ -731,23 +781,18 @@ Type checking
     The ``x`` identifier is bound in the term namespace, but stands for an
     erased, ``forall``-bound type variable.
 
-    Just as with patterns that use the ``type`` herald explicitly, we permit
-    only variables and wildcards in such positions::
-
-      f :: forall a -> ...
-      f x   = ...               -- OK
-      f _   = ...               -- OK
-      f Int = ...               -- illegal to match on a type
-
-    We could, but we do not relax this requirement even when the type is statically known::
+    A similar principle applies to subpatterns in a constructor pattern that
+    correspond to ``forall a ->`` in the type of the data constructor::
 
-      data T a where
-        Typed :: forall a -> a -> T a
+       data Ex where { MkEx :: forall a -> Show a => a -> Ex }
 
-      g :: T Int -> ...
-      g (Typed Int x) = ...   -- Reasonable, but no.
+       f :: Ex -> String
+       f (MkEx a x) = show (x :: a)
+       --     ^^^
+       --     the `a` corresponds to `forall a ->` in the type of `MkEx`
 
-    This is a conservative decision that can be revised in a later proposal.
+    In general, we permit term patterns in positions where type patterns are
+    expected by applying the T2T transformation, see "T2T-Mapping" below.
 
 Namespaces vs Semantics
 ~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
This is an odd, out of place section.  Why not make it item (6) of 7.1?   And please give an example or two.

There is a numbering bug... section numbers jump from 7.1 to 7.4. Peculiar.

Reply:
I agree it's oddly placed. It's doesn't even need to be part of the spec, as it's just an observation. I'll move it to Effect and Interactions.